shell_integration/MacOSX/CMakeLists.txt

```diff
@@ -7,6 +7,9 @@ add_custom_target( legacy_mac_overlayplugin ALL
         OC_SOCKETAPI_TEAM_IDENTIFIER_PREFIX=${SOCKETAPI_TEAM_IDENTIFIER_PREFIX}
     COMMENT building Legacy Mac Overlay icons)
 
+# Contrary to popular belief, this is called like this no matter what theme/OEM.
+set(OC_OEM_SHARE_ICNS "${CMAKE_BINARY_DIR}/src/gui/ownCloud.icns")
```

Comment:
This fails for me with a clean build (used -GNinja, not sure if relevant):
```
cp: /Users/j/dev/client/buildyo/src/gui/ownCloud.icns: No such file or directory
```

src/gui takes care of that icon, but there isn't any dependency given to the build system. There is this code in cmake/modules/AddAppIconMacro.cmake:

```
# Append the icns file to the sources list so it will be a dependency to the
# main target
list(APPEND ${appsources} ${_outfilename}.icns)
```

Maybe you could do something similar on your side?